src/plugins/intel_npu/tests/functional/behavior/npuw/behavior_tests.cpp

```diff
@@ -0,0 +1,261 @@
+// Copyright (C) 2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "behavior_tests.hpp"
+#include "openvino/opsets/opset11.hpp"
+#include "openvino/runtime/exec_model_info.hpp"
+#include "openvino/runtime/internal_properties.hpp"
+#include "openvino/runtime/make_tensor.hpp"
+#include "openvino/runtime/properties.hpp"
+#include "openvino/util/file_util.hpp"
+#include "openvino/util/shared_object.hpp"
+#include "npuw_private_properties.hpp"
+
+using namespace testing;
+using namespace ov::npuw::tests;
+
+std::shared_ptr<ov::Model> BehaviorTestsNPUW::create_example_model() {
+    auto param = std::make_shared<ov::opset11::Parameter>(ov::element::i64, ov::PartialShape{1, 3, 2, 2});
+    param->set_friendly_name("input");
+    auto const_value = ov::opset11::Constant::create(ov::element::i64, ov::Shape{1, 1, 1, 1}, {1});
+    const_value->set_friendly_name("const_val");
+    auto add = std::make_shared<ov::opset11::Add>(param, const_value);
+    add->set_friendly_name("add");
+    auto result = std::make_shared<ov::opset11::Result>(add);
+    result->set_friendly_name("res");
+    return std::make_shared<ov::Model>(ov::ResultVector{result}, ov::ParameterVector{param});
+}
+
+TEST_F(BehaviorTestsNPUW, TestInfrastructureIsCorrect) {
+    // Set expectations first:
+    EXPECT_CALL(*npu_plugin, get_property).Times(AnyNumber());
+    EXPECT_CALL(*npu_plugin, get_property(std::string("AVAILABLE_DEVICES"), _)).Times(1);
+    EXPECT_CALL(*cpu_plugin, get_property).Times(AnyNumber());
+    EXPECT_CALL(*cpu_plugin, get_property(std::string("AVAILABLE_DEVICES"), _)).Times(1);
```

Comment:
Are these mocks or the real plugins tested?

Reply:
They are mock plugins, will rename them, thanks!